binders/kafka-binder/spring-cloud-stream-binder-kafka-reactive/src/main/java/org/springframework/cloud/stream/binder/reactorkafka/SenderResultMessageHandler.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.reactorkafka;
+
+import reactor.core.publisher.Mono;
+import reactor.kafka.sender.SenderResult;
+
+import org.springframework.integration.handler.AbstractReactiveMessageHandler;
+import org.springframework.messaging.Message;
+
+/**
+ * Message handler for {@link SenderResult}s.
+ *
+ * @param <T> the correlation metadata type.
+ *
+ * @author Gary Russell
+ * @since 4.0.3
+ *
+ */
+public abstract class SenderResultMessageHandler<T> extends AbstractReactiveMessageHandler {
```

Comment:
Not sure why do we need this extra adapter. Why just plain `MessageHandler` is not enough?
Or in most cases I guess that is going to be like this:
```
@ServiceActivator(inputChannel = "myRecordMetadataChannel")
void(SenderResult<MyData> result) {
    ...
}
```

Reply:
My thought was this would be convenient to use in the DSL.

```java
IntegrationFlows.from(MessageChannels.flux)
    .handle(new SenderResultMessageHandler<Integer>() {...})
    .get();
```

Reply:
That would work, but it would have to be `.<SenderResult<Integer>>handle(...)`.

Ok; will remove it.

Reply:
Any further motivation to keep this?